thought-machine/please-vscode
# Python debugging

This PR mainly introduces debugging for python.

But also:
- No only Go and Python tests can be debuggable from the source files, but also through the BUILD files.
- Enhances BUILD files to allow targets to be built, run and debugged directly from them.
- Simplifies the debugging experience by not having to care about configuration settings.
- Major refactoring done to allow code to be added more easily and more structured in the future.

**How to test**
- Check the instructions on the new README included in this PR.
- You might have to set `go.delvetool` on the `.plzconfig` of the repo that you are testing against to point to a new version of Delve. We are using an old version internally.

![Screenshot from 2021-10-21 14-13-36](https://user-images.githubusercontent.com/15216823/138928726-bfda6abf-0c8c-461d-b675-f447a7922fa7.png)

![Screenshot from 2021-10-21 14-14-24](https://user-images.githubusercontent.com/15216823/138928587-ccff554b-b6a7-4444-9828-0aab9e1db4ac.png)

![Screenshot from 2021-10-21 14-14-06](https://user-images.githubusercontent.com/15216823/138928824-946432e0-135d-462e-a147-7758fdc52a9d.png)
